### PR TITLE
ENT-3916 - document testing CorDapp upgrades.

### PR DESCRIPTION
**Pull request by tudor-malene**
see https://r3-cev.atlassian.net/browse/ENT-3916

### CODE REVIEW

File: docs/source/upgrading-cordapps.rst [hunk 1]
```diff
@@ -641,3 +641,29 @@ Although not strictly related to versioning, AMQP serialisation dictates that we
...
+To test an implicit upgrade, you must simulate a network that initially has only nodes with the old version, and then nodes gradually transition to the new version.
+Typically, in such a complex upgrade scenario, there must be a deadline by which time all nodes that want to continue to use the CorDapp must upgrade.
+To achieve this, this deadline must be configured in the flow logic which must only use new features afterwards.
+
+This can be simulated with a scenario like this:
+
+1. Write and individually test the new version of the state and contract.
+2. Setup a network of nodes with the previous version. In the simplest form, `deployNodes` can be used for this purpose.
+3. Run some transactions between nodes.
+4. Upgrade a couple of nodes to the new version.
```
Comment (shamsasari) [inline docs/source/upgrading-cordapps.rst]: the wording here could imply new version of the node. Perhaps clarify with "of the CorDapp" at the end

File: docs/source/upgrading-cordapps.rst [hunk 2]
```diff
@@ -641,3 +641,29 @@ Although not strictly related to versioning, AMQP serialisation dictates that we
...
+5. Continue running transactions between various combinations of versions. Also make sure transactions that were created between nodes with the new version
+are  being successfully read by old nodes.
```
Comment (shamsasari) [inline docs/source/upgrading-cordapps.rst]: ditto: old nodes

**Comment by tudor-malene** [comment]
@gssoma , is this what you expected?

**Comment by tudor-malene** [comment]
@shamsasari , ready to merge

**Pull request merged**